src/crawlee/beautifulsoup_crawler/_beautifulsoup_crawler.py

```diff
@@ -181,6 +181,12 @@ async def enqueue_links(
         ) -> None:
             kwargs.setdefault('strategy', EnqueueStrategy.SAME_HOSTNAME)
 
+            if self._max_crawl_depth is not None and context.request.crawl_depth + 1 > self._max_crawl_depth:
+                context.log.info(
+                    f'Skipping enqueue_links for URL "{context.request.url}" due to the maximum crawl depth limit.'
+                )
+                return
+
```

Comment:
1. the repetition in all crawlers is not great
    - it is very easy to overlook this when implementing a new crawler
3. it is also possible to add requests to the queue via `context.add_requests`
    - ideally we should fill in default depth to the requests - it doesn't make much sense to exempt some requests from `max_crawl_depth`
    - if there is no ergonomic way to support max depth in `add_requests`, we should make it clear in the docs that the user has to handle crawl depth manually

Reply:
Yes, I think that the check could go directly to `add_requests` in `BasicCrawler`. Is there any potential issue?